src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1965,8 +1950,13 @@ void PhaseCCP::analyze() {
     _types.map(i, Type::TOP);
   }
 
+  // CCP worklist is placed on a local arena, so that we can allow ResourceMarks on "Compile::current()->resource_arena()".
+  // We also do not want to put the worklist on "Compile::current()->comp_arena()", as that one only gets de-allocated after
+  // Compile is over. The local arena gets de-allocated at the end of its scope.
+  ResourceArea local_arena(mtCompiler);
+  Unique_Node_List worklist(&local_arena);
+
   // Push root onto worklist
-  Unique_Node_List worklist;
   worklist.push(C->root());
   DEBUG_ONLY(Unique_Node_List worklist_verify;)
```

Comment:
Should you put `worklist_verify` to `local_arena` too?

Reply:
@vnkozlov I could do that, but it is not required. The CCP `worklist` gets passed downward much farther, including the  graph walks in `push_child_nodes_to_worklist`. So there it is nice to be able to have `ResourceMarks`. But `worklist_verify` is only modified directly in `PhaseCCP::analyze` and one layer deeper in `PhaseCCP::verify_analyze`. So I would never expect a `ResourceMark` to mess with re-allocation.

So at this point the only reason to add `worklist_verify` to `local_arena` is to ensure it is de-allocated afterward. We could also use a `ResourceMark` and leave it on `Thread::current()->resource_area()`.

Anyway, I'll just move it to `local_arena`, after all it is `DEBUG_ONLY`.

Reply:
ok